zipline/data/_minute_bar_internal.pyx

```diff
@@ -112,7 +128,7 @@ def find_last_traded_position_internal(
 
     minute_pos = int_min(
         find_position_of_minute(market_opens, market_closes, end_minute,
```

Comment:
I see how avoiding the extra `searchsorted` is helpful, but I worry about making the behavior of `find_position_of_minute_minute` adjustable for `ffill` or not; leaving the function as simple as possible.

Another option would be to put the handling of the `NoDataOnDate` in this scope (which is focused on finding the last date.)

Something like:
```
try:
    minute_pos = int_min(
         find_position_of_minute(market_opens, market_closes, end_minute,
                                                minutes_per_day)
except NoDataOnDate:
    prev_close_loc = searchsorted(market_closes, end_minute)
    minute_pos = (prev_close_loc + 1)  * minute_per_day) - 1
```